pkg/agent/controller/networkpolicy/fqdn_test.go

```diff
@@ -584,3 +584,112 @@ func TestSyncDirtyRules(t *testing.T) {
 		})
 	}
 }
+
+func TestOnDNSResponse(t *testing.T) {
+	currentTime := time.Now()
+	tests := []struct {
```

Comment:
I don't see a test case where the existingDNSCache is not impacted by the response, even though it can happen